ebean-ddl-generator/src/test/java/io/ebeaninternal/dbmigration/ddlgeneration/platform/PlatformDdl_AlterColumnTest.java

```diff
@@ -267,79 +398,148 @@ private Column simpleColumn() {
-    assertEquals(oraDdl.useIdentityType(IdType.GENERATOR), IdType.GENERATOR);
-    assertEquals(oraDdl.useIdentityType(IdType.EXTERNAL), IdType.EXTERNAL);
+    softly.assertThat(oraDdl.useIdentityType(null)).isEqualTo(IdType.IDENTITY);
+    softly.assertThat(oraDdl.useIdentityType(IdType.SEQUENCE)).isEqualTo(IdType.SEQUENCE);
+    softly.assertThat(oraDdl.useIdentityType(IdType.IDENTITY)).isEqualTo(IdType.IDENTITY);
+    softly.assertThat(oraDdl.useIdentityType(IdType.GENERATOR)).isEqualTo(IdType.GENERATOR);
+    softly.assertThat(oraDdl.useIdentityType(IdType.EXTERNAL)).isEqualTo(IdType.EXTERNAL);
   }
 
   @Test
   public void useIdentityType_hana() {
 
-    assertEquals(hanaDdl.useIdentityType(null), IdType.IDENTITY);
-    assertEquals(hanaDdl.useIdentityType(IdType.SEQUENCE), IdType.IDENTITY);
-    assertEquals(hanaDdl.useIdentityType(IdType.IDENTITY), IdType.IDENTITY);
-    assertEquals(hanaDdl.useIdentityType(IdType.GENERATOR), IdType.GENERATOR);
-    assertEquals(hanaDdl.useIdentityType(IdType.EXTERNAL), IdType.EXTERNAL);
+    softly.assertThat(hanaDdl.useIdentityType(null)).isEqualTo(IdType.IDENTITY);
+    softly.assertThat(hanaDdl.useIdentityType(IdType.SEQUENCE)).isEqualTo(IdType.IDENTITY);
+    softly.assertThat(hanaDdl.useIdentityType(IdType.IDENTITY)).isEqualTo(IdType.IDENTITY);
+    softly.assertThat(hanaDdl.useIdentityType(IdType.GENERATOR)).isEqualTo(IdType.GENERATOR);
+    softly.assertThat(hanaDdl.useIdentityType(IdType.EXTERNAL)).isEqualTo(IdType.EXTERNAL);
   }
 
   @Test
   public void appendForeignKeySuffix_when_defaults() {
-    assertThat(alterFkey(null, null)).isEqualTo(" on delete restrict on update restrict");
+    softly.assertThat(alterFkey(h2Ddl, null, null)).isEqualTo(" on delete restrict on update restrict");
+    softly.assertThat(alterFkey(pgDdl, null, null)).isEqualTo(" on delete restrict on update restrict");
+    softly.assertThat(alterFkey(mysqlDdl, null, null)).isEqualTo(" on delete restrict on update restrict");
+    softly.assertThat(alterFkey(oraDdl, null, null)).isEqualTo("");
+    softly.assertThat(alterFkey(sqlServerDdl, null, null)).isEqualTo("");
+    softly.assertThat(alterFkey(hanaDdl, null, null)).isEqualTo(" on delete restrict on update restrict");
+    softly.assertThat(alterFkey(db2Ddl, null, null)).isEqualTo(" on delete restrict");
   }
 
   @Test
   public void appendForeignKeySuffix_when_RestrictSetNull() {
-    assertThat(alterFkey("RESTRICT", "SET_NULL")).isEqualTo(" on delete restrict on update set null");
+    softly.assertThat(alterFkey(h2Ddl, "RESTRICT", "SET_NULL"))
+      .isEqualTo(" on delete restrict on update set null");
+
+    softly.assertThat(alterFkey(pgDdl, "RESTRICT", "SET_NULL"))
+      .isEqualTo(" on delete restrict on update set null");
+
+    softly.assertThat(alterFkey(mysqlDdl, "RESTRICT", "SET_NULL"))
+      .isEqualTo(" on delete restrict on update set null");
+
+    softly.assertThat(alterFkey(oraDdl, "RESTRICT", "SET_NULL"))
+      .isEqualTo("");
```

Comment:
@rbygrave  I've extended the test. Please take a look at this, if it is correct for oracle